docs/source/tutorials_source/package/tutorial_simsiam_esa.py

```diff
@@ -0,0 +1,447 @@
+embeddings_2d = (embeddings_2d - m) / (M - m)
+
+
+# %%
+# Let's start with a nice scatter plot of our dataset! The helper function
+# below will create one.
+
+
+def get_scatter_plot_with_thumbnails():
+    """Creates a scatter plot with image overlays.
+    """
+    # initialize empty figure and add subplot
+    fig = plt.figure()
+    fig.suptitle('Scatter Plot of the Sentinel-2 Dataset')
+    ax = fig.add_subplot(1, 1, 1)
+    # shuffle images and find out which images to show
+    shown_images_idx = []
+    shown_images = np.array([[1., 1.]])
+    iterator = [i for i in range(embeddings_2d.shape[0])]
+    np.random.shuffle(iterator)
+    for i in iterator:
+        # only show image if it is sufficiently far away from the others
+        dist = np.sum((embeddings_2d[i] - shown_images) ** 2, 1)
+        if np.min(dist) < 2e-3:
+            continue
+        shown_images = np.r_[shown_images, [embeddings_2d[i]]]
+        shown_images_idx.append(i)
+
+    # plot image overlays
+    for idx in shown_images_idx:
+        thumbnail_size = int(rcp['figure.figsize'][0] * 2.)
+        path = os.path.join(path_to_data, filenames[idx])
+        img = Image.open(path)
+        img = functional.resize(img, thumbnail_size)
+        img = np.array(img)
+        img_box = osb.AnnotationBbox(
+            osb.OffsetImage(img, cmap=plt.cm.gray_r),
+            embeddings_2d[idx],
+            pad=0.2,
+        )
+        ax.add_artist(img_box)
+
+    # set aspect ratio
+    ratio = 1. / ax.get_data_ratio()
+    ax.set_aspect(ratio, adjustable='box')
+
+
+# get a scatter plot with thumbnail overlays
+get_scatter_plot_with_thumbnails()
+
+
+# %%
+# Next, we plot example images and their nearest neighbors (calculated from the
+# embeddings generated above). This is a very simple approach to find more images
+# of a certain type where a few examples are already available. For example,
+# when a subset of the data is already labelled and one class of images is clearly
+# underrepresented, one can easily query more images of this class from the 
+# unlabelled dataset.
+#
+# Let's get to work! The plots are shown below.
+
+example_images = [
+    'S2B_MSIL1C_20200526T101559_N0209_R065_T32TNL/tile_00209.png', # water with ships
+    'S2B_MSIL1C_20200526T101559_N0209_R065_T31TGE/tile_00154.png', # water clear 
+    'S2B_MSIL1C_20200526T101559_N0209_R065_T32TMP/tile_00443.png', # coast line
+    'S2B_MSIL1C_20200526T101559_N0209_R065_T32TMM/tile_00040.png', # land 1
+    'S2B_MSIL1C_20200526T101559_N0209_R065_T32TNL/tile_00556.png', # land 2
+    'S2B_MSIL1C_20200526T101559_N0209_R065_T32SLJ/tile_00527.png', # clouds 1
+    'S2B_MSIL1C_20200526T101559_N0209_R065_T31SGD/tile_01731.png', # clouds 2
+    'S2B_MSIL1C_20200526T101559_N0209_R065_T32SMG/tile_00238.png', # clouds 3
+]
+
+
+def get_image_as_np_array(filename: str):
+    """Returns an image as a numpy array.
+    """
+    img = Image.open(filename)
+    return np.asarray(img)
+
+
```

Comment:
Some comments are very obvious and essentially say the same as the line of code above or below in the same words. Such comments do not add any value to the reader, but are just another thing to read. This applies to following comments: 

>  392: """Returns an image as a numpy array."""
> 402: " get shape of the image "
> 429: "add the subplot"
> 436: "plot the image"
> 441: "disable the axis"

I would rather have fewer and larger comments explaining the idea/summary of what is done than a comment for every line of code.